ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -537,7 +538,10 @@ object BigDiffy extends Command with Serializable {
           .sample(1, head = true).head.getSchema
         implicit val grCoder: Coder[GenericRecord] = Coder.avroGenericRecordCoder(schema)
         val diffy = new AvroDiffy[GenericRecord](ignore, unordered)
-        BigDiffy.diffAvro[GenericRecord](sc, lhs, rhs, avroKeyFn(keys), diffy, schema, ignoreNan)
+        val lhsSCollection = sc.avroFile(lhs, schema)
+        val rhsSCollection = sc.avroFile(rhs, schema)
+        BigDiffy
+          .diff[GenericRecord](lhsSCollection, rhsSCollection, diffy, avroKeyFn(keys), ignoreNan)
```

Comment:
was it intentional to call .diff and not .diffAvro here? I'm not entirely sure what the impact of this change is on the call stack, can you help walk me through it?

Reply:
The scio-avro implementation changed slightly so that `avroFile` either takes a type param that is a sub-type of `SpecificRecord` or no type param and returns `GenericRecord`s.  In order to preserve type safety for users in `diffAvro` we make that operate on SpecificRecords.  So here we need to bypass it and just use diff since we can only operate on `GenericRecord`s from the cli.